doc/LCIO2EDM4hep.md

```diff
@@ -0,0 +1,95 @@
+collections, or potentially even crash.
+
+
+# Integrated use of conversion
+The functions can be used integrated, making the conversion a two step process. Step one is converting the data and step two being the resolving of the relations and filling of subset collection.
+
+There exists a conversion function for every collection that is not of `LCRelation` type (e.g. [`convertReconstructedParticle`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h)). These need to be called before the relations can be handled, since they fill the maps linking the particle in LCIO to their EDM4hep equivalents. Every type has a separate map. The maps are grouped in the `LcioEdmTypeMapping` struct for ease of use. It is also possible to use them directly without the helper struct.
+The order in which the data is converted does not matter because converting data and resolving relations are two differet steps that are carried out in sequence. 
+Subset collections are  handled similar to relations using the fuction [`fillSubset`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h). Internally this simply forwards to [`handleSubsetColl`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h) which handles all the type details and can obviously also be used directly.
+
+The OneToMany and OneToOne Relations can be resolved using [`resolveRelations`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).  There is a resolveRelations function for each type. `LCRelation` only exist in LCIOand there conversion is limited to what is available in EDM4hep. They use the `"FromType"` and `"ToType"` collection parameters to get the necessary type information. 
+
+The AssociationCollections in EDM4hep are then created using [`createAssociations`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h). 
+
+`CaloHitContributions` are treated separately, since they are part of the SimCalorimeterHits in LCIO while being a their own data type in EDM4hep. They are created by [`createCaloHitContributions`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+The EventHeader Colletion can be created using [`EventHeaderCollection`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+Particle IDs are converted during the conversion of the the reconstructed Particle collection.
+
+
+Converting an entire event can be done calling the [`convertEvent`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+## Example for a ReconstructedParticle Collection
+```cpp
+//the structs defined in the header file are used for the maps linking Lcio particles to there EDM counterparts.
+
+#include "k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h"  
+
+auto convertedReconstructedParticleCollection = convertReconstructedParticle(name, LCCollection, typeMapping.recoParticles, typeMapping.particleIDs)
+
```

Comment:
Somewhere it has to be obvious what `LCCollection` is in this case, i.e. explicitly make this a pointer. Also `typeMapping` needs to be declared. On the other hand `name` can be replaced by a string literal. Also the return value can have a much shorter name in this case.